src/lib/io/sol-iio.c

```diff
@@ -88,6 +92,14 @@ struct resolve_absolute_path_data {
     int id;
 };
 
+struct iio_opened_device {
+    struct sol_iio_device *device;
+    struct sol_iio_config *config;
+    int refcount;
+};
+
+static struct sol_ptr_vector iio_opened_devices = SOL_PTR_VECTOR_INIT;
```

Comment:
Maybe a `sol_vector` is easier to use. A good way to know if you need a `sol_ptr_vector` or a `sol_vector` is to check if you need to share pointers to list items. If so, use `sol_ptr_vector`. If not, all you need is an internal list that will always be iterated when needed, a `sol_vector` should suffice.

For instance, the `channels` of a device are stored on a `sol_ptr_vector`, as we give pointers to `channel` to users. As `sol_vector` internal pointers may change, they are not fit for this task.

On the other hand, the instructions to 'address a device' are stored on a `sol_vector`, as pointer to instructions are not shared - we always iterate the `sol_vector` to use them.

Let me know if you had any doubts.